File: dart/lib/src/models/list_tags_result.dart
//
//  rerout
//  list_tags_result.dart
//
//  Official Dart SDK for the Rerout branded-link API.
//
//  Created by Ngonidzashe Mangudya on 20/05/2026.
//  Copyright (c) 2026 Codecraft Solutions
//  Licensed under the MIT License
//  https://codecraftsolutions.co.za
//

import 'package:meta/meta.dart';
import 'package:rerout/src/models/tag_summary.dart';

/// The result of listing the project's tags.
@immutable
class ListTagsResult {
  /// Creates a [ListTagsResult].
  const ListTagsResult({required this.tags});

  /// Parses the listing from the API JSON envelope.
  factory ListTagsResult.fromJson(Map<String, dynamic> json) {
    final tags = (json['tags'] as List<dynamic>? ?? const <dynamic>[])
        .map((e) => TagSummary.fromJson(e as Map<String, dynamic>))
        .toList(growable: false);
    return ListTagsResult(tags: tags);
  }

  /// The project's tags, each carrying its live link count.
  final List<TagSummary> tags;
}

File: java/src/main/java/co/rerout/sdk/model/ListTagsResult.java
/*
 * rerout-java — Official Java SDK for the Rerout branded-link API.
 *
 * Copyright (c) 2026 Codecraft Solutions
 * Licensed under the MIT License — https://codecraftsolutions.co.za
 */

package co.rerout.sdk.model;

import java.util.Collections;
import java.util.List;

/** The tags returned by a {@code Tags.list()} call, each with its link count. */
public final class ListTagsResult {

    private final List<TagSummary> tags;

    /**
     * Creates a {@code ListTagsResult}.
     *
     * @param tags the project's tags; a {@code null} value is normalised to an
     *             empty list
     */
    public ListTagsResult(List<TagSummary> tags) {
        this.tags = tags == null ? Collections.emptyList() : tags;
    }

    /** {@return the project's tags with their live link counts; never {@code null}} */
    public List<TagSummary> getTags() {
        return tags == null ? Collections.emptyList() : tags;
    }
}
